src/components/Tree/Tree.cy.tsx

```diff
@@ -34,37 +66,231 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).first().realMouseDown().realMouseMove(0, 40).realMouseUp();
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('should trigger onDrop on a level 1 item as a controlled component', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(4).realMouseDown().realMouseMove(0, -40).realMouseUp();
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('should not trigger onDrop when moving to an not allowed inside item', () => {
+        const onDropStub = cy.stub().as('onDropStub');
 
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(6).realMouseDown().realMouseMove(40, -40).realMouseUp();
+        cy.get('@onDropStub').should('not.have.been.called');
+    });
+
+    it('allows adding an item dynamically to the Tree', () => {
+        const newNode: TreeItemMock = {
+            id: '4',
+            label: 'New Added Item',
+        };
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.push(newNode);
+            rerender(<TreeComponent />);
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength + 1);
+        });
+    });
+
+    it('allows removing an item dynamically to the Tree', () => {
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.pop();
+            rerender(<TreeComponent />);
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength - 1);
+        });
+    });
+
+    it('navigates correctly with the keyboard', () => {
+        cy.mount(<TreeComponent />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID)
+            .first()
+            .focus()
+            .realPress('Space')
+            .realPress('Tab')
+            .realPress('Space')
+            .realPress('ArrowDown')
+            .realPress('ArrowDown')
+            .realPress('ArrowDown')
+            .realPress('ArrowDown')
+            .realPress('ArrowRight')
+            .realPress('ArrowUp');
         cy.get(TREE_ITEM_ID).should('have.length', 14);
+        cy.get(TREE_ITEM_ID).eq(4).should('have.focus');
     });
+});
 
-    it('calls the Tree `onSelect` callback', () => {
-        const onSelectStub = cy.stub().as('onSelectStub');
-        cy.mount(<TreeComponent onSelect={onSelectStub} />);
+describe('Adding / Removing item dynamically', () => {
```

Comment:
These test cases are the same as above, right?

Reply:
oh, yes... they were moved to its own section but forgot to remove them from the previous section